src/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -214,22 +214,14 @@ async Task<MsalAccessTokenCacheItem> ITokenCacheInternal.FindAccessTokenAsync(Au
 
             if (requestParams.AuthorityInfo != null)
             {
-                var instanceDiscoveryMetadataEntry = await GetCachedOrDiscoverAuthorityMetaDataAsync(
-                    requestParams.AuthorityInfo.CanonicalAuthority,
-                    requestParams.RequestContext).ConfigureAwait(false);
+                InstanceDiscoveryMetadataEntry instanceDiscoveryMetadataEntry =
+                    await ServiceBundle.InstanceDiscoveryManager.GetMetadataEntryAsync(
+                        new Uri(requestParams.AuthorityInfo.CanonicalAuthority),
+                        requestParams.RequestContext)
+                    .ConfigureAwait(false);
 
-                environmentAliases.UnionWith(GetEnvironmentAliases(
-                    requestParams.AuthorityInfo.CanonicalAuthority,
-                    instanceDiscoveryMetadataEntry));
-
-                if (requestParams.AuthorityInfo.AuthorityType == AuthorityType.Adfs)
-                {
-                    preferredEnvironmentAlias = requestParams.AuthorityInfo.CanonicalAuthority;
-                }
-                else if (requestParams.AuthorityInfo.AuthorityType != AuthorityType.B2C)
-                {
-                    preferredEnvironmentAlias = instanceDiscoveryMetadataEntry.PreferredCache;
-                }
+                environmentAliases.UnionWith(instanceDiscoveryMetadataEntry.Aliases);
```

Comment:
why do you need this unionwith?  environmentAliases is always empty.  this can just be a set.